### PR TITLE
feat: Fetch token from url on start

### PR DESCRIPTION
**Pull request by aviadl**
This way we can have a flow that will start with verify token 
related to https://github.com/descope/etc/issues/2288

### CODE REVIEW

File: packages/web-component/src/lib/descope-wc/DescopeWc.ts [hunk 1]
```diff
@@ -139,12 +139,14 @@ class DescopeWc extends BaseDescopeWc {
...
-              exchangeCode: code,
-              idpInitiated: true,
-            }
-          : undefined;
+        const inputs = {};
+        if (code) {
+          inputs['exchangeCode'] = code;
+          inputs['idpInitiated'] = true;
+        }
+        if (token) {
```
Comment (orius123) [inline packages/web-component/src/lib/descope-wc/DescopeWc.ts:147]: Can you add a test?

File: packages/web-component/src/lib/descope-wc/DescopeWc.ts [hunk 2]
```diff
@@ -322,8 +324,8 @@ class DescopeWc extends BaseDescopeWc {
     // If there is a start screen id, next action should start the flow
     // But if oidcIdpStateId is not empty, this optimization doesn't happen
     if (showFirstScreenOnExecutionInit(startScreenId, oidcIdpStateId)) {
-      stepStateUpdate.next = (interactionId, inputs) =>
-        this.sdk.flow.start(
+      stepStateUpdate.next = (interactionId, inputs) => {
+        return this.sdk.flow.start(
```
Comment (orius123) [inline packages/web-component/src/lib/descope-wc/DescopeWc.ts:328]: In what case is it needed?

File: packages/web-component/src/lib/descope-wc/DescopeWc.ts
Comment (asafshen) [inline packages/web-component/src/lib/descope-wc/DescopeWc.ts:1]: We also need to handle cleanup of the token in this case
image

need to examine the code above to see where it make sense to do it

**Review by asafshen — COMMENTED** [review]
code looks good, we need to ensure cleanup is done as well

File: packages/web-component/src/lib/descope-wc/DescopeWc.ts (hunk 2 above)
Comment (asafshen) [reply packages/web-component/src/lib/descope-wc/DescopeWc.ts:328]: if we want the invite to the flow to start with a screen - we will need it

Comment (aviadl) [reply packages/web-component/src/lib/descope-wc/DescopeWc.ts:328]: When flow starts with a screen for some reason
But we still want to take the token from the URL

File: packages/web-component/src/lib/descope-wc/DescopeWc.ts
Comment (aviadl) [reply packages/web-component/src/lib/descope-wc/DescopeWc.ts:1]: @asafshen dont understand your comment

Comment (asafshen) [reply packages/web-component/src/lib/descope-wc/DescopeWc.ts:1]: currently, after we use token (e.i after it is sent to `update` request - we clean the token from the component state

we should do the same to not pass token again in next request

**Review by asafshen — DISMISSED** [review]

**Review by asafshen — APPROVED** [review]

**Pull request merged**